### PR TITLE
Update gambit and gerbil

### PR DESCRIPTION
**Pull request by fare**
###### Motivation for this change

Use some recent capabilities of gerbil, that themselves require a recent unreleased gambit.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/gambit/default.nix [hunk 1]
```diff
@@ -1,31 +1,71 @@
...
+    mkdir -p boot &&
+    cp -rp ${bootstrap}/. boot/. &&
+    chmod -R u+w boot &&
+    rm -f boot/gsc/*.scm &&
+    cp -a gsc/*.scm boot/gsc/ &&
+    (cd boot/gsc && make bootclean ) &&
+    (cd boot && ./configure && make bootstrap ) &&
+    cp -a boot/gsc-boot* ./ &&
+    make -j2 from-scratch &&
+    make check
```
Comment (0xABAB) [inline pkgs/development/compilers/gambit/default.nix]: I think there is a separate phase for this.

**Review by 0xABAB — CHANGES_REQUESTED** [review]
Put make check in a different phase.

**Comment by fare** [comment]
Indeed. I added `doCheck = true;` instead.

**Comment by fare** [comment]
Yet another update, to enable various FFI libraries in gerbil.

File: pkgs/development/compilers/gerbil/default.nix [hunk 2]
```diff
@@ -20,6 +33,7 @@ stdenv.mkDerivation rec {
   '';
 
   buildPhase = ''
+    substituteInPlace "src/std/build-config.ss" --replace '#f)' '#t)'
```
Comment (0xABAB) [inline pkgs/development/compilers/gerbil/default.nix]: This needs a comment and seems fragile for updates.

**Review by 0xABAB — APPROVED** [review]
Thanks, now the code is obvious.

**Comment by fare** [comment]
Ping. What's the next step?

**Comment by 0xABAB** [comment]
Some options:
* You are patient and just wait. 
* You can ping someone who is a 'Member' and not a mere 'Contributor' (like I am) and who has been recently active.
* You can look on IRC whether you see someone speak with a nickname matching one of the member's nicknames.
* You can send an e-mail to the mailing list

The strategy I would recommend to you is to use the code you have produced yourself and then just hope that it will be merged and see PRs as a kind of asynchronous system. 

This could be handled better, sure.

**Comment by fare** [comment]
NB: Meanwhile I'll keep updating those commits, mostly to take advantage of gambit and gerbil updates.

Please make sure to review again before you merge, though it should be mostly version changes and build script tweaks.

**Comment by 0xABAB** [comment]
@fare I think you should also use the `nativeBuildInputs` attribute https://github.com/NixOS/nixpkgs/issues/19370. See the packaging of `mit-scheme` as an example. More information here: https://github.com/NixOS/nixpkgs/issues/1915.

**Comment by fare** [comment]
While gambit *can* be built with a cross-compiler, the current package won't build if you do. Moreover, you'd need to be able to tell that that the dependency on the bootstrap package was a native dependency, not a target dependency.

As for gerbil, I suppose it also could be built with a cross-compiler, but then I might need both a native gambit *and* a target gambit, *and* a cross gambit. That's getting complicated.

**Review by 0xABAB — APPROVED** [review]

**Comment by 0xABAB** [comment]
@7c6f434c or @Mic92 Please merge this.

**Comment by Mic92** [comment]
merged both in f96e4edb177728dfe3565825a706c96ff706ff63

**Comment by 0xABAB** [comment]
@Mic92 The diff was way bigger. It seems you missed some files.

**Pull request closed without merging**